packages/server/lib/video_capture.js

```diff
@@ -272,76 +273,79 @@ module.exports = {
 
     let total = null
 
-    return new Promise((resolve, reject) => {
-      debug('processing video from %s to %s video compression %o',
-        name, cname, videoCompression)
+    return fs.mkdirp(path.dirname(cname)).then(() => {
```

Comment:
@jennifer-shehane this is the only line I changed, the rest is indentation

Reply:
Although there's nothing wrong with your patch, I don't see how this could fix the issue you're seeing. The compressed video is output to the exact same folder as the uncompressed video, so it should be impossible to reach this point if the parent directory doesn't exist.

We're you able verify this patch fixes the issue you're having?

Reply:
@Bkucera I have not verified that this patch fixes things, however I am seeing issues related to folders not existing in CI. do you have some thoughts on what that might be in relation to outside of this?

Reply:
@Bkucera @daviddungio mentioned in #17455 it is something related to running in parallel and that seems to be more in-line with what I'm seeing, so maybe this fix might be unrelated.

Reply:
This is the error output from Jenkins

```
====================================================================================================


  (Run Starting)


  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐

  │ Cypress:    6.9.1                                                                                                                           │

  │ Browser:    Chrome 92 (headless)                                                               │

  │ Specs:      16 found (###/###/###.spec.js, #                                                   │

  │             #/#/#-#.spec.js, #/#/#                                                                       │

  │             #-#.spec.js, #/#/#-#.spec.js, #                                                    │

  │             /#...)                                                                                                        │

  │ Params:     Tag: stage, Group: chrome, Parallel: true                                          │

  │ Run URL:    https://dashboard.cypress.io/projects/#/runs/#                                     │

  └────────────────────────────────────────────────────────────────────────────────────────────────┘



────────────────────────────────────────────────────────────────────────────────────────────────────

                                                                                                    

  Running:  #/#/#-#.spec.js                          (3 of 16)

	Just tests tagged with: smoke

Warning: We failed to record the v


  Initiative Performance Tests

the exit code.


Error: ffmpeg exited with code 1: /opt/src/cypress/ui/cypress/videos/#/#-#-#.spec.js.mp4: No such file or directory


    at ChildProcess.<anonymous> (/opt/cypress/6.9.1/Cypress/resources/app/packages/server/node_modules/fluent-ffmpeg/lib/processor.js:182:22)

    at ChildProcess.emit (events.js:315:20)

    at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)


Error: ffmpeg exited with code 1: /opt/src/cypress/ui/cypress/videos/#/#-#-#.spec.js.mp4: No such file or directory


    at ChildProcess.<anonymous> (/opt/cypress/6.9.1/Cypress/resources/app/packages/server/node_modules/fluent-ffmpeg/lib/processor.js:182:22)

    at ChildProcess.emit (events.js:315:20)

    at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)


Error: ffmpeg exited with code 1: /opt/src/cypress/ui/cypress/videos/#/#-#-#.spec.js.mp4: No such file or directory


    at ChildProcess.<anonymous> (/opt/cypress/6.9.1/Cypress/resources/app/packages/server/node_modules/fluent-ffmpeg/lib/processor.js:182:22)

    at ChildProcess.emit (events.js:315:20)

    at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
```